src/schemathesis/_hypothesis.py

```diff
@@ -114,17 +118,23 @@ def get_case_strategy(endpoint: Endpoint) -> st.SearchStrategy:
 
     Path & endpoint are static, the others are JSON schemas.
     """
+    strategies = {}
     static_kwargs = {"path": endpoint.path, "method": endpoint.method, "base_url": endpoint.base_url}
     try:
-        strategies = {
-            "path_parameters": from_schema(endpoint.path_parameters)
-            .filter(filter_path_parameters)  # type: ignore
-            .map(quote_all),  # type: ignore
-            "headers": from_schema(endpoint.headers).filter(is_valid_header),  # type: ignore
-            "cookies": from_schema(endpoint.cookies),
-            "query": from_schema(endpoint.query),
-            "form_data": from_schema(endpoint.form_data),
-        }
+        for parameter in PARAMETERS:
+            if getattr(endpoint, parameter):
+                if parameter == "path_parameters":
+                    strategies[parameter] = (
+                        from_schema(endpoint.path_parameters)
+                        .filter(filter_path_parameters)  # type: ignore
+                        .map(quote_all)  # type: ignore
+                    )
+                elif parameter == "headers":
+                    strategies[parameter] = from_schema(endpoint.headers).filter(is_valid_header)  # type: ignore
+                else:
+                    strategies[parameter] = from_schema(getattr(endpoint, parameter))  # type: ignore
```

Comment:
could you, please define `value = getattr(endpoint, parameter)` and use it in all branches ?